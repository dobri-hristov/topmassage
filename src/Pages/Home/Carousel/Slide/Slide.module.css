.imgContainer {
    width: 100%;
    height: 85vh;
}

.imgContainer img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    filter: brightness(0.8);
}

.legend {
    transition: all .5s ease-in-out;
    position: relative;
    color: #fff;
    text-align: center;
    transition: opacity .35s ease-in-out;
}

.legend>* {
    margin: 8px;
}

.btn {
    border-radius: 15px;
    border: none;
    padding: 15px 20px;
    color: white;
    background: #673AB7;
    font-size: 16px;
}

.btn:hover {
    cursor: pointer;
    box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.75);
    transition: all 0.3s ease-in-out;
    color: black;
}

.legend1, .legend3, .legend2 {
    width: 30%;
    border-radius: 20px;
}

.legend1 {
    bottom: 45%;
    left: 2%;
}

.legend2 {
    bottom: 95%;
    left: 5%;
}

.legend3 {
    width: 50%;
    bottom: 95%;
    left: 45%;
}

@media screen and (max-width: 600px),
screen and (max-height: 500px) {
    .imgContainer img {
        filter: brightness(0.4);
    }

    .legend1, .legend3, .legend2 {
        width: 100%;
    }

    .legend1 {
        bottom: 95%;
        left: 0%;
    }

    .legend2 {
        left: 0%;
    }

    .legend3 {
        width: 100%;
        bottom: 95%;
        left: 0%;
    }
}

@media screen and (max-width: 600px) {
    .imgContainer {
        height: 60vh;
    }
}