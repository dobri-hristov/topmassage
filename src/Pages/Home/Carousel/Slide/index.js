import React from 'react'
import { useHistory } from "react-router-dom";
import styles from './Slide.module.css'

const Slide = ({ image, title, info, legendClass }) => {
    const history = useHistory()

    return (
        <div className={styles.imgContainer}>
            <img src={image} alt="massage information" />
            <div className={`${styles.legend} ${styles[legendClass]}`}> 
                {
                    title
                        ? <h1>{title}</h1>
                        : null
                }
                <p>{info}</p>
                <button onClick={() => history.push('/reservation')} className={styles.btn}>Запази час</button>
            </div>
        </div>
    )
}

export default Slide
