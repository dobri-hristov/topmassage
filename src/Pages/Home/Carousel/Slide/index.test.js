import React from 'react'
import TestRenderer from 'react-test-renderer'
import Slide from './index.js'

describe('Home page/Carousel/Slide component', () => {
    test('should render correctly', () => {
        const component = TestRenderer.create(
            <Slide
                image=''
                title='Класически масаж'
                info='Класически масаж информация'
                legendClass='legend1' />
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})