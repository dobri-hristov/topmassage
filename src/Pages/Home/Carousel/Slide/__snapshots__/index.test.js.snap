// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page/Carousel/Slide component should render correctly 1`] = `
<div
  className="imgContainer"
>
  <img
    alt="massage information"
    src=""
  />
  <div
    className="legend legend1"
  >
    <h1>
      Класически масаж
    </h1>
    <p>
      Класически масаж информация
    </p>
    <button
      className="btn"
      onClick={[Function]}
    >
      Запази час
    </button>
  </div>
</div>
`;
